### PR TITLE
Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby on Rails/The Asset Pipeline

### LINKED ISSUE #22939: Add missing Knowledge Checks, Learning Outcomes, and Knowledge Check links to Ruby on Rails/The Asset Pipeline
Related to #22470

This task is to go through the The Asset Pipeline lesson in the Ruby on Rails course and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id='example-knowledge-check'>consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class='knowledge-check-link' href='#example-knowledge-check'>How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class='knowledge-check-link' href='#lorem-ipsum-dolor-sit-amet'>How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### PR DESCRIPTION
**Pull request by pudu87**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [x] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
- [x] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1.Describe the changes made and include why they are necessary or important:

- Adjust learning outcomes
- Add knowledge checks

#### 2. Related Issue

closes #22939

### CODE REVIEW

File: rails_programming/rails_basics/asset_pipeline.md [hunk 1]
```diff
@@ -168,3 +167,12 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Ryan Bates' asset pipeline Railscast](http://railscasts.com/episodes/279-understanding-the-asset-pipeline?view=asciicast)
 * [Schneems on the Asset Pipeline](https://www.youtube.com/watch?v=FYdBpNUVxuI)
 * [Stack Overflow on Escaping HTML in Rails](http://stackoverflow.com/questions/692921/rails-how-to-html-encode-escape-a-string-is-there-a-built-in)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class="knowledge-check-link" href="https://guides.rubyonrails.org/asset_pipeline.html#main-features">How does "asset concatenation" improves faster loading of your app?</a>
```
Comment (leosoaivan) [inline rails_programming/rails_basics/asset_pipeline.md:174]: Could we switch out "improves faster" with an alternative? Something like "optimize"?

**Review by leosoaivan — CHANGES_REQUESTED** [review]

File: rails_programming/rails_basics/asset_pipeline.md (hunk 1 above)
Comment (pudu87) [reply rails_programming/rails_basics/asset_pipeline.md:174]: Sure, no problem.

File: rails_programming/rails_basics/asset_pipeline.md [hunk 2]
```diff
@@ -168,3 +167,12 @@ This section contains helpful links to other content. It isn't required, so cons
 * [Ryan Bates' asset pipeline Railscast](http://railscasts.com/episodes/279-understanding-the-asset-pipeline?view=asciicast)
 * [Schneems on the Asset Pipeline](https://www.youtube.com/watch?v=FYdBpNUVxuI)
 * [Stack Overflow on Escaping HTML in Rails](http://stackoverflow.com/questions/692921/rails-how-to-html-encode-escape-a-string-is-there-a-built-in)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class="knowledge-check-link" href="https://guides.rubyonrails.org/asset_pipeline.html#main-features">How does "asset concatenation" optimizes loading of your app?</a>
```
Comment (leosoaivan) [inline rails_programming/rails_basics/asset_pipeline.md:174]: `optimizes` -> `optimize`

**Review by leosoaivan — CHANGES_REQUESTED** [review]

**Comment by leosoaivan** [comment]
@pudu87 thanks for making the change. I have one final grammatical fix I'd like you to push up, and then this should be good to go!

File: rails_programming/rails_basics/asset_pipeline.md (hunk 2 above)
Comment (pudu87) [reply rails_programming/rails_basics/asset_pipeline.md]: Done!

**Review by leosoaivan — APPROVED** [review]

**Pull request merged**